conformance/lang/expressions/regexp/regexp_semantics.balt

```diff
@@ -0,0 +1,719 @@
+    io:println(t.get(re `ABC`).re == re `ABC`); // @output true
+
+    io:println(t.get(re `DEF`).re is string:RegExp); // @output true
+    io:println(t.get(re `DEF`).re == re `DEF`); // @output true
+}
+
+Test-Case: output
+Description: Test 1 for regular expression as list member.
+Labels: RegExp, list-constructor-expr
+
+function init() {
+    string:RegExp[] regArr = [re `ABC`, re `([A-D]+|[H-Z]*)|C{2,9}`, re `[a-z]`];
+    io:println(regArr.length().toString()); // @output 3
+    io:println(regArr[0] == re `ABC`); // @output true
+    io:println(regArr[1] == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+    io:println(regArr[2] == re `[a-z]`); // @output true
+}
+
+Test-Case: output
+Description: Test 2 for regular expression as list member.
+Labels: RegExp, list-constructor-expr
+
+function init() {
+    string:RegExp[3] regArr = [re `ABC`, re `([A-D]+|[H-Z]*)|C{2,9}`, re `[a-z]`];
+    io:println(regArr.length().toString()); // @output 3
+    io:println(regArr[0] == re `ABC`); // @output true
+    io:println(regArr[1] == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+    io:println(regArr[2] == re `[a-z]`); // @output true
+}
+
+Test-Case: output
+Description: Test 3 for regular expression as list member.
+Labels: RegExp, list-constructor-expr
+
+function init() {
+    string:RegExp[*] regArr = [re `ABC`, re `([A-D]+|[H-Z]*)|C{2,9}`, re `[a-z]`];
+    io:println(regArr.length().toString()); // @output 3
+    io:println(regArr[0] == re `ABC`); // @output true
+    io:println(regArr[1] == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+    io:println(regArr[2] == re `[a-z]`); // @output true
+}
+
+Test-Case: output
+Description: Test assignability of regular expression array.
+Labels: RegExp, list-constructor-expr
+
+function init() {
+    string:RegExp[3] regArr1 = [re `ABC`, re `([A-D]+|[H-Z]*)|C{2,9}`, re `[a-z]`];
+    string:RegExp[] regArr2 = regArr1;
+    io:println(regArr2.length().toString()); // @output 3
+    io:println(regArr2[0] == re `ABC`); // @output true
+    io:println(regArr2[1] == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+    io:println(regArr2[2] == re `[a-z]`); // @output true
+
+    string:RegExp[*] regArr3 = [re `ABC`, re `([A-D]+|[H-Z]*)|C{2,9}`, re `[a-z]`];
+    string:RegExp[3] regArr4 = regArr3;
+    io:println(regArr4.length().toString()); // @output 3
+    io:println(regArr4[0] == re `ABC`); // @output true
+    io:println(regArr4[1] == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+    io:println(regArr4[2] == re `[a-z]`); // @output true
+
+    string:RegExp[] regArr5 = regArr3;
+    io:println(regArr5.length().toString()); // @output 3
+    io:println(regArr5[0] == re `ABC`); // @output true
+    io:println(regArr5[1] == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+    io:println(regArr5[2] == re `[a-z]`); // @output true
+}
+
+Test-Case: output
+Description: Test regular expression as tuple member.
+Labels: RegExp, tuple-type-descriptor
```

Comment:
Should be `tuple-type`
missing `tuple-rest-descriptor`, `list-constructor-expr`